domain-models-runtime/src/main/java/org/folio/rest/impl/TenantAPI.java

```diff
@@ -309,54 +201,115 @@ public String sqlFile(String tenantId, boolean tenantExists, TenantAttributes te
    */
   @Validate
   @Override
-  public void postTenant(TenantAttributes tenantAttributes, Map<String, String> headers,
-      Handler<AsyncResult<Response>> handler, Context context)  {
+  public void postTenant(TenantAttributes tenantAttributes, RoutingContext routingContext, Map<String, String> headers,
+                         Handler<AsyncResult<Response>> handler, Context context)  {
+    postTenant(true, tenantAttributes, routingContext, headers, handler, context);
+  }
+
+  private void postTenant(boolean async, TenantAttributes tenantAttributes, RoutingContext routingContext, Map<String, String> headers,
+                          Handler<AsyncResult<Response>> handler, Context context)  {
 
     String tenantId = TenantTool.tenantId(headers);
-    log.info("sending... postTenant for " + tenantId);
-    if (tenantAttributes != null) {
-      log.debug("upgrade from " + tenantAttributes.getModuleFrom() + " to " + tenantAttributes.getModuleTo());
+    String id = UUID.randomUUID().toString();
+    TenantJob job = new TenantJob();
+    job.setId(id);
+    job.setTenant(tenantId);
+    job.setTenantAttributes(tenantAttributes);
+    job.setComplete(false);
+
+    jobs.put(id, job);
+    String location = (routingContext != null ? routingContext.request().uri() : "") + "/" + id;
```

Comment:
POST https://example.com:9876/_/tenant
with a
Location: /12345
reply will result in the wrong
https://example.com:9876/12345
```suggestion
    String location = (routingContext != null ? routingContext.request().uri() : "/_/tenant") + "/" + id;
```
https://tools.ietf.org/html/rfc7231#section-7.1.2
https://tools.ietf.org/html/rfc3986#section-4.2
https://tools.ietf.org/html/rfc3986#section-5.4

Reply:
No. the uri() returns ` /_/tenant`.

Reply:
The routingContext == null case is only for non-HTTP testing. There is obviously no test for Location in that case.

Reply:
Actually, I'll see if I can pass some (non-null) routingContext in tests. No need to have production code that checks for null routingContext anyway.

Reply:
Yes, the only change that I suggest is the `routingContext == null` case.

Reply:
ok. Since you insist on hard-coding the path in the case that is not used in production code, I'll just switch to not use RoutingContext. That also avoids the generate_routing context property which, as I've mentioned before, is evil :-)